pypdf/_page.py

```diff
@@ -1691,7 +1691,8 @@ def process_operation(operator: bytes, operands: List) -> None:
                                     )
                                 text = ""
                         elif (
-                            abs(delta_y) < f * 0.3
+                            rtl_dir is False
+                            and abs(delta_y) < f * 0.3
```

Comment:
I'm confused : no delta_y leading to add a space has a mean with rtl_dir == True ?

also this should not be completed to cover the other orientations?

Reply:
It's adding irrelevant white space in some cases. I'll be inspecting the white space estimation further and try to figure out the exact cause.

And you're correct, it should be completed to cover the other orientations too so I'll be taking that into consideration.

Reply:
@pubpub-zz 

Here's how's the file with only odd "h" glyph ligature is extracted:
`assert reader.pages[0].extract_text() == "حَبيبي h"`

Will translates to:
`AssertionError: assert ' يبيبَحh' == 'حَبيبي h'`

Two issues are noticed:
- Wrong empty space placement
- Wrong "Fatha" diacritic placement

The one to multi-chars test will fail because it's a glyph that contains both RTL and LTR characters (Arabic+Latin and also White Space). Copying it from Acrobat and inspecting the clipboard RTF content demonstrate that the resulting text is gibberish, non coherent, and non readable.

This is non standard and this case will never exists in real life (I mean the case of a glyph with bidirectional text). I can't see how this test case could help in testing and debugging the real-life solution.

My suggestion is to ignore this non-standard case for now, fix the Arabic text extraction on a standard and common case, and maybe bring back the odd habibi.pdf after making sure that the extraction is accurate on standard Arabic and Bidi text.

Let me know what are your thoughts so I can proceed.